arangod/Aql/IResearchViewExecutor.h

```diff
@@ -322,8 +319,6 @@ class IndexReadBuffer {
       if (newMemoryUsage != tracked) {
         if (newMemoryUsage > tracked) {
           _memoryTracker.increase(newMemoryUsage - tracked);
-        } else {
```

Comment:
If this branch cannot be reached, we should change the nested if conditions from
```
if (newMemoryUsage != tracked) {
  if (newMemoryUsage > tracked) {
```
to just
```
if (newMemoryUsage > tracked) {
```

Reply:
Fixed